### PR TITLE
expose lookup tables

### LINKED ISSUE #112: expose tables in API
I'm writing a RAM stub in assembly for an embedded system that I'm going to use to quickly CRC regions of flash memory as part of the process of flashing an MCU. I was going to use this crate to generate CRC lookup tables that I would also load via the debug probe at a known address to accelerate this, but the table-access functionality is private &mdash; I just need the whole table that the crate already generates.

I can work around this -- I've just copy-pasted the `crc8` and `crc8_table` functions into my crate, but for my use-case it'd be a nice, useful addition to this crate's API.

Would you accept a PR to expose the generated tables and table generation functions? I'd like to:

- Either make the `table` field of `Crc` public or add a `pub const fn Crc::table(&self) -> I::Table`
- Make the `table` module and its functions public

### PR DESCRIPTION
**Pull request by BKSalman**
resolves #112 

any comments?

### CODE REVIEW

**Comment by akhilles** [comment]
LGTM. The new `Implementation` API was just merged (https://github.com/mrhooray/crc-rs/pull/115). Do you mind rebasing? Sorry for the churn.

**Comment by BKSalman** [comment]
Will do, np

**Comment by BKSalman** [comment]
@akhilles should I make it like this?
```rust
    pub const fn table(&self) -> <Table<1> as Implementation>::Data<u8> {
        self.data
    }
```

File: src/crc128/bytewise.rs [hunk 1]
```diff
@@ -35,6 +35,10 @@ impl Crc<u128, Table<1>> {
         let value = init(self.algorithm, initial);
         Digest::new(self, value)
     }
+
+    pub const fn table(&self) -> <Table<1> as Implementation>::Data<u128> {
+        self.data
```
Comment (akhilles) [inline src/crc128/bytewise.rs:40]: Should return a reference of the table to avoid copying.

**Review by akhilles — CHANGES_REQUESTED** [review]

**Comment by BKSalman** [comment]
I will be off for holidays, and won't be able to commit for a couple of days, I will try to tho since it's just a small change

File: src/crc128/bytewise.rs (hunk 1 above)
Comment (BKSalman) [reply src/crc128/bytewise.rs]: @akhilles done

**Comment by BKSalman** [comment]
Thank you :)

**Pull request merged**